### PR TITLE
Correctly handle an array of data_bag paths

### LINKED ISSUE #191: Support data_bag_path arrays
Since chef12, `Chef::Config[:data_bag_path]` supports arrays.

When using knife vault is solo mode, the data_bag_path is `File.join`-ed (https://github.com/chef/chef-vault/blob/master/lib/chef-vault/item_keys.rb#L68) which generates invalid paths.

For instance in knife.rb:
```
data_bag_path [
  File.join(chef_repo_path, 'site-data_bags'),,
  File.join(chef_repo_path, 'data_bags')
]
```

the computed data_bag_path is a concatenation of those paths + the name of the vault data bag.

I don't know what the behavior could be:
- selecting the first one
- adding a new configuration in chef-vault to override the chef data_bag_path

### PR DESCRIPTION
**Pull request by thommay**
This was introduced in Chef 12. Fixes #191

### CODE REVIEW

**Comment by kamaradclimber** [comment]
thanks @thommay for this quick patch

**Comment by kamaradclimber** [comment]
I've just tested it with success

**Comment by thommay** [comment]
@kamaradclimber excellent, thanks for the confirmation.

File: lib/chef-vault/mixins.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class ChefVault
+  module Mixins
+    # In Chef 12, Chef Solo can have an array of data_bag paths, rather
+    # than just a string. To cope with that, we'll:
+    #  1. Look for an existing data bag item in any of the configured
+    #     paths and use that by preference
+    #  1. Otherwise, just use the first location in the array
+    def find_solo_path(item_id)
+      if Chef::Config[:data_bag_path].kind_of?(Array)
+        path = Chef::Config[:data_bag_path].select { |dir|
```
Comment (stevendanna) [inline lib/chef-vault/mixins.rb]: Minor style suggestion: I like to use the `find` method when I just need the first item that matches, saving you the call #first.

**Comment by stevendanna** [comment]
:+1:

**Pull request merged**